### DISCUSSION TITLE
TypeScript support

### DISCUSSION THREAD

**Discussion by yuvaraj-thiagarajan** [bangle-io/bangle.dev#166] [Ideas]
Presently, I do not see any TypeScript declaration file (.d.ts file) in the repository. Since TypeScript adoption is increasing, it would be nice to have TypeScript delaration files or @types/* npm packages for @bangle.dev/core, @bangle.dev/react and other modules. Hope this will get prioritized and available for use soon!

**Comment by kepta** [comment]
Having typescript is great, but I am pretty strapped with time and I am currently focusing on making the library stable and feature complete. If would be happy to accept PRs converting the module into typescript, but currently for me feature development and stability is a priority.

**Reply by lucywang000** [reply]
I'll take a look at this task.

**Comment by kepta** [comment]
We have kicked off typescript support with 2 modules converted to TS (collab-client & collab-server), with that setting up the basic infra, we can convert more to typescript.

**Reply by lucywang000** [reply]
Awesome @kepta ! 

Which package do you think could be the next candidate for TS conversion?

**Reply by kepta** [reply]
I think the smaller packages (other than `bangle.dev/react` or `bangle.dev/core`) will be easy to start as they are small and we can stub the common dependencies like done https://github.com/bangle-io/bangle.dev/blob/master/collab-server/__tests__/manager.test.ts#L1 . Once most of the leaf packages are converted we can then move the bigger packages to typescript.

**Reply by lucywang000** [reply]
Cool, I'll start with the tooltip package.

**Comment by kepta** [comment]
This was done.
